include/robot_interfaces/loggable.hpp

```diff
@@ -0,0 +1,34 @@
+///////////////////////////////////////////////////////////////////////////////
+// BSD 3-Clause License
+//
+// Copyright (C) 2018-2019, Max Planck Gesellschaft
+// Copyright note valid unless otherwise stated in individual files.
+// All rights reserved.
+///////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include <Eigen/Eigen>
+
+#include <robot_interfaces/robot_backend.hpp>
+#include <robot_interfaces/robot_data.hpp>
+#include <robot_interfaces/robot_frontend.hpp>
+
+/**
+* @brief Contains definitions of the methods to be implemented by the n joint
+* robot data types. The idea is that any external block such as the logger can
+* extract the information about the fields of the Observation and Action structures
+* using these methods. 
+* **/
+
+namespace robot_interfaces
+{
+class Loggable
+{
+public:
+
+  virtual std::vector<std::string> get_name() = 0;
+  virtual std::vector<std::vector<double>> get_data() = 0;
```

Comment:
Sort of random fly by comment: Should there also be timesteps corresponding to the recorded data?

Reply:
The timeseries already provides time information (the "time index" as well as a timestamp), so in general there should be no need to have it here.

Reply:
What's the information that should be represented in the `Loggable` object? For me, the contained data should also be the time index. This is not included in this struct yet.

I can see that the information is on the timeseries object. However, do you intend to forward this time information to the logger when using the Loggable interface?

Reply:
The information that should be represented in the 'Loggable' object- the names of the fields in the 'Loggable' structure returned by the 'get_name()' method, and the fields themselves returned by 'get_data()'. This is being done so that we don't explicitly have to know what's inside the object. 

The time information is available to the logger by getting the timestamps associated with each element. So, there's no connection with Loggable for the time information.

I am also not sure I understood your question correctly, happy to discuss offline.

Reply:
Let's discuss offline. I think this is a more fundamental design discussion and better done on a whiteboard :)

Reply:
Yes, great :)